### ISSUE TITLE
Build Instructions

### ISSUE THREAD

**Issue by mottosso** [PySide/pyside2#2]
Possible premature, but if possible, it would be amazing to follow along and perhaps test things out as things are progressing.

I've got a number of PyQt5 projects I'd be very interested in porting.

**Comment by IronManMark20** [comment]
I am concerned that Python 2.7 won't be usable, unless compiled with something other than VS 2008. None of the QT VS plugins are for VS 2008. So either you use MinGW, or Python 3 (which may be a good idea anyway). I guess you could compile Python 2 with a newer version of Python.

I am trying to follow the [instructions for Qt4](https://pyside.readthedocs.org/en/latest/building/windows.html). Of course, I have to change a few things. I am currently stuck as it is failing to build the docs for shiboken.

**Comment by mottosso** [comment]
I think Python 2 would ultimately have to be supported, as this effort is primarily a push from the visual effects industry and the Autodesk ecosystem (?) for which 2.6-2.7 is ubiquitous.

**Comment by ctismer** [comment]
IronMan: be aware: you will need to build your own Qt5!
I just got QtCore to compile.
Needed to disable QtConcurrent for now.
QtGui barfs at "'PyCapsule_New' : cannot convert parameter 1 from 'WId' to 'void *'"
But that is Python3 specific.

**Comment by IronManMark20** [comment]
>IronMan: be aware: you will need to build your own Qt5! 

Oh, that would explain the errors. Thanks. I think I'll wait until tomorrow when you're putting out a README. Thanks for your great work!!

**Comment by maxime-esa** [comment]
I am out of the office until 01/09/2015.




Note: This is an automated response to your message  "Re: [pyside2] Build
Instructions (#2)" sent on 15/08/2015 20:27:08.

This is the only notification you will receive while this person is away.
This message and any attachments are intended for the use of the addressee or addressees only.
The unauthorised disclosure, use, dissemination or copying (either in whole or in part) of its
content is not permitted.
If you received this message in error, please notify the sender and delete it from your system.
Emails can be altered and their integrity cannot be guaranteed by the sender.

Please consider the environment before printing this email.

**Comment by maxime-esa** [comment]
I am out of the office until 01/09/2015.




Note: This is an automated response to your message  "Re: [pyside2] Build
Instructions (#2)" sent on 15/08/2015 20:33:23.

This is the only notification you will receive while this person is away.
This message and any attachments are intended for the use of the addressee or addressees only.
The unauthorised disclosure, use, dissemination or copying (either in whole or in part) of its
content is not permitted.
If you received this message in error, please notify the sender and delete it from your system.
Emails can be altered and their integrity cannot be guaranteed by the sender.

Please consider the environment before printing this email.

**Comment by maxime-esa** [comment]
I am out of the office until 01/09/2015.




Note: This is an automated response to your message  "Re: [pyside2] Build
Instructions (#2)" sent on 15/08/2015 20:43:17.

This is the only notification you will receive while this person is away.
This message and any attachments are intended for the use of the addressee or addressees only.
The unauthorised disclosure, use, dissemination or copying (either in whole or in part) of its
content is not permitted.
If you received this message in error, please notify the sender and delete it from your system.
Emails can be altered and their integrity cannot be guaranteed by the sender.

Please consider the environment before printing this email.

**Comment by maxime-esa** [comment]
I am out of the office until 01/09/2015.




Note: This is an automated response to your message  "Re: [pyside2] Build
Instructions (#2)" sent on 15/08/2015 20:16:49.

This is the only notification you will receive while this person is away.
This message and any attachments are intended for the use of the addressee or addressees only.
The unauthorised disclosure, use, dissemination or copying (either in whole or in part) of its
content is not permitted.
If you received this message in error, please notify the sender and delete it from your system.
Emails can be altered and their integrity cannot be guaranteed by the sender.

Please consider the environment before printing this email.

**Comment by empyrical** [comment]
This is why you make sure your autoresponder is properly configured, folks :p 

Making sure @maxime-esa is pinged so they see this later

**Comment by maxime-esa** [comment]
I am out of the office until 01/09/2015.




Note: This is an automated response to your message  "Re: [pyside2] Build
Instructions (#2)" sent on 15/08/2015 21:26:49.

This is the only notification you will receive while this person is away.
This message and any attachments are intended for the use of the addressee or addressees only.
The unauthorised disclosure, use, dissemination or copying (either in whole or in part) of its
content is not permitted.
If you received this message in error, please notify the sender and delete it from your system.
Emails can be altered and their integrity cannot be guaranteed by the sender.

Please consider the environment before printing this email.

**Comment by IronManMark20** [comment]
Why is it still replying? It is way after the 9th of January.

**Comment by maxime-esa** [comment]
I am out of the office until 01/09/2015.




Note: This is an automated response to your message  "Re: [pyside2] Build
Instructions (#2)" sent on 15/08/2015 22:42:38.

This is the only notification you will receive while this person is away.
This message and any attachments are intended for the use of the addressee or addressees only.
The unauthorised disclosure, use, dissemination or copying (either in whole or in part) of its
content is not permitted.
If you received this message in error, please notify the sender and delete it from your system.
Emails can be altered and their integrity cannot be guaranteed by the sender.

Please consider the environment before printing this email.

**Comment by kirbyfan64** [comment]
Uhhhh...

...I think something's broken...

**Comment by The-Compiler** [comment]
@IronManMark20 See the date format in "Note: This is an automated response to your message" - this probably means 1st of September.

I'll contact GitHub support to see if they can do anything against this, as it's definitely annoying.

This could be easily avoided if @maxime-esa wasn't using an auto-replier which is just... completely dumb.

**Comment by empyrical** [comment]
I think it means that he's gone until September 1, so this guy won't be shutting off his auto responder for quite some time.

@ctismer maybe you can block @maxime-esa for now and let him know he can get unblocked when he gets back and turns the auto responder off?

**Comment by kirbyfan64** [comment]
I guess the auto-replier is just for general email messages, but he didn't realize that it would reply to the GitHub notification messages, too.

**Comment by The-Compiler** [comment]
It really shouldn't if it was at least a little bit sane, as GitHub does their best to prevent this kind of thing:

```
Return-Path: <noreply@github.com>
Precedence: list
X-Auto-Response-Suppress: All
```

But as said, it seems to be completely stupid and broken. I wonder what mail provider/software that is.

**Comment by ctismer** [comment]
I guess I will trash the whole issue tracker and start over.
This thingie is not at all helpful for me, but boring rubbish.

@empyrical no sorry, I am working hard at PySide2, and I don't care about people who are wasting my time through their sloppiness..

**Comment by maxime-esa** [comment]
I am out of the office until 01/09/2015.




Note: This is an automated response to your message  "Re: [pyside2] Build
Instructions (#2)" sent on 15/08/2015 23:36:35.

This is the only notification you will receive while this person is away.
This message and any attachments are intended for the use of the addressee or addressees only.
The unauthorised disclosure, use, dissemination or copying (either in whole or in part) of its
content is not permitted.
If you received this message in error, please notify the sender and delete it from your system.
Emails can be altered and their integrity cannot be guaranteed by the sender.

Please consider the environment before printing this email.

**Comment by maxime-esa** [comment]
I am out of the office until 01/09/2015.




Note: This is an automated response to your message  "Re: [pyside2] Build
Instructions (#2)" sent on 15/08/2015 23:48:34.

This is the only notification you will receive while this person is away.
This message and any attachments are intended for the use of the addressee or addressees only.
The unauthorised disclosure, use, dissemination or copying (either in whole or in part) of its
content is not permitted.
If you received this message in error, please notify the sender and delete it from your system.
Emails can be altered and their integrity cannot be guaranteed by the sender.

Please consider the environment before printing this email.

**Comment by maxime-esa** [comment]
I am out of the office until 01/09/2015.




Note: This is an automated response to your message  "Re: [pyside2] Build
Instructions (#2)" sent on 16/08/2015 00:18:56.

This is the only notification you will receive while this person is away.
This message and any attachments are intended for the use of the addressee or addressees only.
The unauthorised disclosure, use, dissemination or copying (either in whole or in part) of its
content is not permitted.
If you received this message in error, please notify the sender and delete it from your system.
Emails can be altered and their integrity cannot be guaranteed by the sender.

Please consider the environment before printing this email.

**Comment by maxime-esa** [comment]
I am out of the office until 01/09/2015.




Note: This is an automated response to your message  "Re: [pyside2] Build
Instructions (#2)" sent on 15/08/2015 23:56:06.

This is the only notification you will receive while this person is away.
This message and any attachments are intended for the use of the addressee or addressees only.
The unauthorised disclosure, use, dissemination or copying (either in whole or in part) of its
content is not permitted.
If you received this message in error, please notify the sender and delete it from your system.
Emails can be altered and their integrity cannot be guaranteed by the sender.

Please consider the environment before printing this email.

**Comment by maxime-esa** [comment]
I am out of the office until 01/09/2015.




Note: This is an automated response to your message  "Re: [pyside2] Build
Instructions (#2)" sent on 15/08/2015 23:37:26.

This is the only notification you will receive while this person is away.
This message and any attachments are intended for the use of the addressee or addressees only.
The unauthorised disclosure, use, dissemination or copying (either in whole or in part) of its
content is not permitted.
If you received this message in error, please notify the sender and delete it from your system.
Emails can be altered and their integrity cannot be guaranteed by the sender.

Please consider the environment before printing this email.

**Comment by thopiekar** [comment]
Just as an addition to this discussion. I would be very interested into instructions how to make an android app. I already started making a qt5 recipe for a toolchain: https://github.com/kivy/python-for-android/pull/439
Would be great, if you could tell how to setup the Android project and here to copy all the different files, like {installdir}/plugins.

Thanks!

**Comment by IronManMark20** [comment]
@thopiekar that won't happen for a while. There is a ways to go before any builds on Desktop are usable, so I wouldn't get your hopes up too high, it will take time. Once it is finished, I see no reason why it wouldn't be about as simple as for PyQt5, but only once PySide2 is at least beta status (which it isn't to my knowledge).

**Comment by thopiekar** [comment]
No problem :) I guessed it will take some time. Just wanted to make you notice to care about that and create more attention on our project.
Maybe some people would be interested helping out and share knowledge about setting everything up.

Thank you for your quick reply :)

**Comment by ctismer** [comment]
@IronManMark20 you don't need to create your own build on windows any longer.
There is a branch "qt5-win-fix" which has the changed flags which the standard Qt5 uses.
I will remove this difference soon by a script that checks the given settings via qmake.

In any case, there is a large list of things to be done. Especially, it makes little sense
to try to "fix" most tests, because some things need to be re-written, completely.
I will write about the current status, soon.

**Comment by IronManMark20** [comment]
Excellent news! I'm going to try to compile a debug 2.7 today or tomorrow, I'll let you know how it goes. Thanks so much for your awesome work.

**Comment by ctismer** [comment]
As an addition: PySide2 builds with Qt5.4 or less, only.
I have tried to move on to Qt5.5, but no luck, it _cannot_ work without changing the shiboken parser! :-(
You can read about that on the branch "qt5.4-qt5.5-migrate". I wanted to show that that one is eventually
quite easy to do, but it was not - the first syntax change since qt4.x because of a new, important
macros. And macros are defined as extra tokens (yes, not kidding, "the parser" is quite a hack...)

**Comment by IronManMark20** [comment]
Good to know thanks. Also, for future reference, I am going to try out [this build of Py2.7](http://www.p-nand-q.com/python/building-python-27-with-visual_studio.html), since he seems to have debug working. I'll see what I can do.

**Comment by IronManMark20** [comment]
Hmmm. My build failed with a linking error (1120) for QtCore.pyd. I will post a gist with the full debug list soon (maybe in the morning). It was 22% through building QtCore. 

A snippet of the full output:

    [ 22%] Building CXX object PySide/QtCore/CMakeFiles/QtCore.dir/glue/qeasingcurve_glue.cpp.obj
    qeasingcurve_glue.cpp
    Linking CXX shared module ..\QtCore.pyd
       Creating library QtCore.lib and object QtCore.exp
    
    qtcore_module_wrapper.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: int __thiscall QString::toWCharArray(unsigned short *)const " (__imp_?toWCharArray@QString@@QBEHPAG@Z) referenced in function "struct _object * __cdecl QString_CppToPython_QString(void const *)" (?QString_CppToPython_QString@@YAPAU_object@@PBX@Z)
    
    ..\QtCore.pyd : fatal error LNK1120: 1 unresolved externals
    
    LINK failed. with 1120
    NMAKE : fatal error U1077: 'C:\cmake\bin\cmake.exe' : return code '0xffffffff'
    Stop.
    NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN\nmake.exe"' : return code '0x2'
    Stop.
    NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN\nmake.exe"' : return code '0x2'
    Stop.
    error: Error compiling pyside

**Comment by ctismer** [comment]
Then I guess you used
`you don't need to create your own build on windows any longer.`
but you ignored my hint
`There is a branch "qt5-win-fix" which has the changed flags which the standard Qt5 uses.`

Also, your script was through 22% of building PySide. QtCore was complete.

**Comment by IronManMark20** [comment]
I did checkout qt5-win-fix. when i run the setup.py, it seems to update submodules, reverting back to the qt5 branch. i will try removing this and rebuild. On the 22% yes it is for the whole thing I miss wrote that.

**Comment by ctismer** [comment]
Ah, sorry, I might have forgotten to mention the necessary option `--ignore-git` which I probably mentioned somewhere else on the mailing list. Sorry about that, I really need to collect information in a central place, now.

I have a little problem with putting docs into pyside2, because there are three and possibly more branches in the future.
Would it be ok if I put a small readme, that points to a more stable repos for adding developer infos? Where would I put that stuff, maybe into a folder of pyside-setup2?

**Comment by IronManMark20** [comment]
That makes sense. I would change the readme for that, since it still carries the qt4 readme.

I just ran it. I am past QtCore (it built!) I am going through QtGui, but I have to leave. I will let you know how it goes. I can write up steps for Windows if you want, once I am done building.

**Comment by The-Compiler** [comment]
> I have a little problem with putting docs into pyside2, because there are three and possibly more branches in the future.
> Would it be ok if I put a small readme, that points to a more stable repos for adding developer infos? Where would I put that stuff, maybe into a folder of pyside-setup2?

GitHub also has a Wiki feature, maybe put it there and link from the readme?

**Comment by ctismer** [comment]
@The-Compiler not bad, true. But then I would want to save a copy of that Wiki in a repos, because I don't like to have that info bound to a certain site (GitHub). Is there a doable way to do that?

**Comment by The-Compiler** [comment]
I'd argue that's a worse solution than just having a local clone of the wiki repo. It's basically just a link, except that's in the repo instead of the README - but the data still won't be saved except if you clone the submodule - and at this point, why not just clone the wiki repo?

**Comment by ctismer** [comment]
Ok, this is very new to me, and I was't aware that the wiki is a repo in its own. Thanks, I will follow your advice, blindly.
Is that the wiki of pyside2, then, and not of pyside-setup2?
I'm asking because there is also shiboken2, which actually lacks a lot of information, too. But the split into shiboken and pyside is IMHO already unhealthy, because they really belong together and are sharing problems:

At the moment, I was working in the `qt5.4-to-qt5.5-migrate` branch. This branch is stuck, because I need to change the shiboken parser, in order to support a new macro in pyside. So where should I document that?

Thanks for advice ;-)

**Comment by The-Compiler** [comment]
It's hard to say... In general, every repo/project has its own wiki, yeah.

Maybe you just want to document everything under `pyside2`, and link to that from the shiboken repo too? The pyside repo is probably where I'd search them intuitively.

**Comment by ctismer** [comment]
Ok, I'll do that.
So the pyside2 wiki is a sub-repos of pyside2. What happens with different branches, then?
Ah, probably this is just a matter of propagating fresh linkage after every wiki change into every active branch of pyside2. No big deal, yeah ;-)

**Comment by IronManMark20** [comment]
Hmm. So it failed building. 46% through pyside. It is progress, but :(  

[ 46%] Building CXX object PySide/QtGui/CMakeFiles/QtGui.dir/PySide/QtGui/qtgui_module_wrapper.cpp.obj
qtgui_module_wrapper.cpp
C:\pyside-setup2\pyside_build\py2.7-qt5.4.2-32bit-release\pyside\PySide\QtGui\PySide\QtGui\qtgui_module_wrapper.cpp(665) : error C2664: 'PyCObject_FromVoidPtr' : cannot convert parameter 1 from 'WId' to 'void *'
        Conversion from integral type to pointer type requires reinterpret_cast, C-style cast or function-style cast
NMAKE : fatal error U1077: 'C:\PROGRA~2\MICROS~4.0\VC\bin\cl.exe' : return code '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN\nmake.exe"' : return code '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN\nmake.exe"' : return code '0x2'
Stop.
error: Error compiling pyside

**Comment by thopiekar** [comment]
Just want to note you that creating submodules will break code imports for
launchpad.net as it uses bzr for hosting code and bzr has no support for
converting git submodules. So if you plan making nightly builds at
launchpad, which would be great for testing the latest code, please don't
create submodules.

Regards

**Comment by ctismer** [comment]
@thopiekar hmm, well...
Under these circumstances, it is probably best to stick with the one parent module pyside-setup2 where pyside2 is already a member of, and use that wiki.

**Comment by thopiekar** [comment]
@ctismer: Right, pyside-setup2 is already broken for imports.

**Comment by ctismer** [comment]
@IronManMark20 well, I just built with python 2.7 and 3.4 on OSX,
so this seems to be something special on Windows with python 2.7.
It is quite likely to be a visible thing, and probably in a python3 conditional branch. Can you please check your generated file, or upload it for me? I'd like to avoid building 2.7 on VS2010.

**Comment by ctismer** [comment]
@IronManMark20 ok, this is a python 2.7 effect on Windows.
I installed your python2.7 and fixed this by changing the XML for windows QtGui.

**Comment by IronManMark20** [comment]
@ctismer, thanks. I chose to use the VS 2010 out of necessity. If you look at the Qt MSVC versions, the earliest is 2010. Python 2.7 was compiled with 2008, and it is bad to mix Visual Studio versions. Therefore one is essentially forced to use the VS 2010 version. We could have people build their own Qt5s (or provide our own?), but using this pre-compiled build seemed easier. I pulled your updates and am compiling again, fingers crossed this works.

**Comment by ctismer** [comment]
About VS versions: No, it is possible to have multiple versions installed, and choosing the right one by selecting the appropriate batch script.
I am just lazy and refuse to install more that VS 2010.
Note that you cannot go up from Python 3.4, unless you install VS 2015!

About fingers crossed: This has to work, or something is wrong. I don't expect anything but success.

We need to proceed and fix things. There are five serious flaws, will explain them, a bit later.

**Comment by IronManMark20** [comment]
You can have multiple versions installed (I must have all of them by now), but using a library compiled for VS2010, and Python 2.7, which was compiled with VS2008 is not okay. [Read more here](https://docs.python.org/2/extending/windows.html#a-cookbook-approach). Of note `You will still need the C compiler that was used to build Python`. 

Anyway, it built! The main problem I'm having is that importing PySide didn't work. It did not like the `from . import _utils`. The PEP 8 discourages relative imports. I cd'ed into the PySide directory, which is a temporary fix, but obviously, this will not work for all solutions.

**Comment by empyrical** [comment]
One thing to make note of - the Chromium in QtWebengine won't work with vs2010 - iirc the oldest it will work with is vs2013.

**Comment by IronManMark20** [comment]
@mottosso if you want, here is a _rough_ guide for building:

1. Get OpenSSL, cmake, and Qt 5.4.x (seems Christian and I are both using 5.4.2). This is very platform dependent. It should be very similar to [these instructions](https://pyside.readthedocs.org/en/latest/building/index.html), except you should get Qt from [the archives](https://download.qt.io/archive/qt/5.4/5.4.2/)
2. `git clone --recursive https://github.com/PySide/pyside-setup2.git`
3. `cd pyside-setup2`
4. `python.exe setup.py install --qmake=/path/to/bin/qmake --cmake=/path/to/bin/cmake --openssl=/path/to/openssl/bin --ignore-git`. Note that the `qmake` and `cmake` are looking for something to execute, but openssl is just looking for the bin folder. 

Some notes for Windows. Between steps 3 and 4, you should `cd sources && cd pyside2` so that you can `git checkout qt5-win-fix`, which allows you to compile PySide2.

Also, @ctismer, would it not make sense to rename the package name to PySide2 or something like that? I expect people will want to have them side-by-side.

@empyrical, good to know, but I doubt that will be too important, as it isn't breaking compatibility beyond the given Qt (at least for the 2010 version).

**Comment by ctismer** [comment]
Yes, but this irrelevant right now. 
The matter is to get an acceptable prototype up and running, and we are
pretty far from that:

There are seriously flawed things internally, and without community
help, I have no chance to grt it done. 

Sent from my Ei4Steve

**Comment by empyrical** [comment]
That's definitely true - just thought I'd mention that because it bit me once when I was trying to get PyQt5 working on a 2010-based python3

**Comment by ctismer** [comment]
Hey, thank you!

A few things: you use 'install' instead of 'develop'. What makes the difference?

On qt5-win-fix:
This will be gone by tomorrow. I think to add my script to 
pyside-tools, and enforce it's existance. 

Then, after that, I think to change the repo structure. I will rename 'master' to 'qt4' and 'qt5' to 'master'. 
I think this internediate state is now over and no more needed. 

Then, I want to write about PySide2 on some lists, and ask for help. 
Help may be by code or by money/contracts. 

The lists will be Python, Python-dev, PySide, PyDide-dev, Python-Announce. 

Any questions/suggestions/distractions about that?

Cheers -- Chris. 


Sent from my Ei4Steve

**Comment by ctismer** [comment]
Hey,

if you have real experience with PyQt5, please try to share it with
me/us. I am more lije a beginner on that, although with (30 years of) background. 

Sent from my Ei4Steve

**Comment by The-Compiler** [comment]
python-dev is a bad idea - it's a high-signal mailinglist about the development of Python itself, so I don't think such a message will be appropriate/welcome there.

I'm also not sure if it'll be approved for python-announce, which is moderated and AFAIK for *releases* only.

I'm using PyQt5 myself for 2 years or so now - what kind of experience are you looking for?

**Comment by ctismer** [comment]
@The-Compiler I am actually looking for some useful input, as you submitted right now.
I am not sure where to write what, but I did this many times for Stackless Python, and
therefore I'm not fearing bad replies.

To be more explicit: Is it true that I can/should not announce an Alpha in the public, or what are you trying to suggest?

Cheers - Chris

**Comment by IronManMark20** [comment]
@ctismer I don't think I'll be able to help directly (ie bug fixing), but I can help with secondary things like continuous integration and working with the public.

Here are a few starter ideas:

* I can post something on /r/python. There are about 110000 people subscribed to /r/python, and the last post announcing that this project was being worked was upvoted 200+ times (and read by many more). This may be a way to spread the word. 

* Continuous Integration. Since we are targeting several operating systems, it would probably be a good idea to set up Continuous integration. I can set up travis ci for linux, appveyor for windows, and I can look into building on Mac OS X. 

* I can also start working on documentation, build instructions and other guides.

All of these will allow you and others to focus more on bug crushing, and getting to beta (and beyond). 

also, on the mailing lists, most of the release posts (most are past v1.0) are on python.announce.

**Comment by The-Compiler** [comment]
> To be more explicit: Is it true that I can/should not announce an Alpha in the public, or what are you trying to suggest?

Not at all, no - I think it's a good idea to get the word out! I'm just saying `python-dev` definitely isn't the right place, as that's offtopic there:

> On this list the key Python developers discuss the future of the language and its implementation. Topics include Python design issues, release mechanics, and maintenance of existing releases.

**Comment by empyrical** [comment]
@ctismer I'll definitely pipe up if I have anything else to add, I'm still pretty green to this sort of thing

**Comment by ctismer** [comment]
Hey empyrical,

I just realized what this means to me:

I'm supposed to create a working PySide2 which can replace PySide, with no new bugs, and on VS2010. 

If I understand this right, then the customer does not really use 
QtWebEngine, which is great for me :-)

Cheers -Chris

Sent from my Ei4Steve

**Comment by empyrical** [comment]
Oh, I see. Basically have an API similar to the Qt4 PySide, but using Qt5 instead.

QtWebEngine has a very different API from QtWebkit and it's not exactly a drop-in replacement for QtWebkit - its api is mostly asynchronous for starters. It needs the QtWebChannel module which uses websockets for its inter-process communication, bit of a learning curve! Bummer that QtWebkit is starting to get phased out of official qt releases though: https://www.mail-archive.com/development@qt-project.org/msg18866.html - but it wouldn't surprise me if it ended up getting community maintenance for a while

**Comment by ctismer** [comment]
@empyrical Ah, I understand. No, they want me to support the old QtWebKit, so my comment was pretty useless. Supporting QtWebEngine is most likely out of scope of this project for now.

Cheers - Chris

**Comment by empyrical** [comment]
Also, do you plan on trying to upstream PySide2 when it's stable? If so then folks with code contributions will need to sign the Qt CLA. I don't know if they plan on staying maintainers of PySide though or if they'll do what they did with Jambi and pass maintainership to community members - last "official" word I've seen from them was from 2013

**Comment by mottosso** [comment]
@IronManMark20 Sorry for the late reply, and thanks! I'm enjoying the resulting conversation and will take a look next time I'm in Qt town.

**Comment by IronManMark20** [comment]
@mottosso Welcome. Glad I could help. Good luck building.

**Comment by ctismer** [comment]
Hi Guys,

FYI: Today, I was able to fix the parser. So you will get PySide2 for Qt5.5, very soon.

I am about to submit some intro to certain lists. Before doing that, I will rename the repository
paths in the following way:

"origin" from 4.8: becomes "qt4"
"qt5" from the new dev becomes "origin"
everything else stays as it is.

Right now, I am in the transition phase, with "qt4.8" and "origin", which is "qt5.4".
There is also "qt5.5". This will not be renamed, before it is valid.

So again, here the expected mappings:

origin  -> qt4
qt5    -> origin
qt5.5 -> qt5.5
etc.

**Comment by kirbyfan64** [comment]
Do you mean `master`, since `origin` is the URL of the GitHub repo...

**Comment by ctismer** [comment]
Yes, sure. I'm always messing thus stuffbup.

**Comment by empyrical** [comment]
Started a rough initial dependencies page (still need to make a windows section):

https://github.com/PySide/pyside-setup2/wiki/Dependencies

**Comment by IronManMark20** [comment]
@empyrical, I can start on the Windows dependencies. It built for me with cmake, qt, and openssl

**Comment by IronManMark20** [comment]
@ctismer,

what will happen to qt5-win-fix? Will that be merger into master and qt5.5?

**Comment by ctismer** [comment]
@IronManMark20 I just merged qt5-win-fix.
This was an oversight from me, that I should have recognized, earlier.

Today, I wanted to explain the Windows problem to Roman, the maintainer of pyside-setup (Qt4),
when I realized that I never checked this setting on Qt4.8. I installed it and compared the qmake.conf
files in mkspecs\win32-msvc2010, and in fact, this flag was changed, and I was wrong here.
Fixed!

**Comment by IronManMark20** [comment]
@ctismer good to hear! I'll run through the build process again and write it up in the wiki later.

**Comment by ctismer** [comment]
@IronManMark20 The 5.5 branch is almost ready, so maybe you want to wait a few hours until I merge it?

**Comment by IronManMark20** [comment]
I'll be busy till then Anyway.  I'll add it in 8 hours.

**Comment by ctismer** [comment]
Hi Guys,

I just managed to get PySide building with Qt5.5 . So this was one major deliverable. 

I now have a few hairy problems left, and I'm pretty sure that I am not efficient at 
solving them all. 

What do you suggest how to explain thr problens, and where?

I think to put this into a Wiki, probably into a PySide2 wiki on Github. Is that ok?

What do you think?

**Comment by thopiekar** [comment]
If there are problems why not adding it as issues to github? Issues can
also be todos I think ;)

However, congrats getting it working! Is it only working on Windows now?

**Comment by ctismer** [comment]
Hi Thomas,

no offense, but you gave the best example why I'm so careful with
asking for "help". 

When everybody asks me how much how and where works, then I'm 
spending all my time on helping the helpers. 

What I can really use are those who find out by reading all my checkins and write something useful. 

Cheers - Chris

Sent from my Ei4Steve

**Comment by thopiekar** [comment]
Got it. Sorry, for asking something stupid like that. Sounds like it's time
for me to make nightly builds on launchpad.net.

Again, many thanks for your work!

**Comment by ctismer** [comment]
I will talk to you in private :-)

**Comment by ctismer** [comment]
On 17.09.15 23:19, Thomas-Karl Pietrowski wrote:
> Got it. Sorry, for asking something stupid like that. Sounds like it's
> time
> for me to make nightly builds on launchpad.net.
>
> Again, many thanks for your work!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PySide/pyside2/issues/2#issuecomment-141231018>.
>

Hi friend,

can I please get your email address?

cheers -- tismer@stackless.com

-- 
Christian Tismer             :^)   tismer@stackless.com
Software Consulting          :     http://www.stackless.com/
Karl-Liebknecht-Str. 121     :     https://github.com/PySide
14482 Potsdam                :     GPG key -> 0xFB7BEE0E
phone +49 173 24 18 776  fax +49 (30) 700143-0023

**Comment by IronManMark20** [comment]
I added to the Windows section for dependencies. It isn't perfect, but it works.

**Comment by ctismer** [comment]
@IronManMark20 
> Also, @ctismer, would it not make sense to rename the package name to PySide2 or something like that? I expect people will want to have them side-by-side.

Ok, you proposed that the package name should be pyside2 ?
I think it is time for that, right. Is that name ok, or should we repeat PyQt4/PyQt5 maybe?

**Comment by thopiekar** [comment]
Same question for shiboken: Should it also be co-installable? At the moment
shiboken2 e.g. produces a "shiboken" executable, which makes installations
side-by-side not possible.

**Comment by IronManMark20** [comment]
@ctismer , @thopiekar Well, from what I have read on the original post on /r/Python, people in general prefered a 4/5 in the naming, so that they could easily tell the difference. However, it would keep closer to internal naming to call it 2.0, since the last Qt4 was 1.2.2. For shiboken, if we want side-by side, then yes, we will want different names.

**Comment by ctismer** [comment]
@thopiekar @IronManMark20 Hmm, yes, true. Shiboken exists also as its own project (I tend to often ignore that), therefore I agree that not only the repositories, but also the projects should be renamed.

Actually, shiboken2 instead of shiboken is a good way to keep people from just picking pyside2 and trying to build.

Stays the decision about the name... Maybe we should take the "2" for now, because people already know about it?

**Comment by thopiekar** [comment]
Sounds good. I even started making Ubuntu packaging using "shiboken2" as
package name. :+1:

**Comment by ctismer** [comment]
Ok, then I should be consequent and rename the project folders also when packaging etc., and the cmake files use that as project name, too. Correct?

**Comment by IronManMark20** [comment]
pyside2-setup makes more sense to me.

**Comment by ctismer** [comment]
Then we have the following repositories:
PySide/pyside2-setup
PySide/pyside2
PySide/pyside2-examples
PySide/pyside2-tools

When should I do these renames? I think to start with the pyside2-examples, because that hurts nobody...

**Comment by IronManMark20** [comment]
Yep. Perhaps create shadows of the old ones pointing to the new ones. Will I am updating the readmes for all the repos I might as well.

**Comment by ctismer** [comment]
how do I create a shadow?

**Comment by IronManMark20** [comment]
Just create a repo with the old names, so that Github won't 404 people, then in a readme we direct them to the new names.

**Comment by ctismer** [comment]
Also, we need to decide where the wiki lives, pyside2-setup or pyside2. Right now, there is one too much ;-)

Ah, thank you!

**Comment by ctismer** [comment]
ok, I will start top-level right now with pyside2-setup.

**Comment by IronManMark20** [comment]
I think pyside2-setup, since that is where we are sending everyone. Also, issues aren't exactly made for chat, may I recommend that development chat move to [gitter](https://gitter.im/home)?

And awesome! I will fix my PR are broken now. (oops).

**Comment by thopiekar** [comment]
I would place the wiki into pyside2, because pyside2-setup won't be the
only way to install pyside on the target. E.g. debian packaging does not
need pyside2-setup at all, because it has got it's own instructions.

PS: Just to remember: I agree with every solution as long as submodules are
not placed into pyside2, pyside2-tools, pyside2-examples and shiboken2.
Already said this will break code imports in Launchpad.net and kill future
nightly/testing builds there ;)

**Comment by ctismer** [comment]
ok, the wiki goes in pyside2, this is ok.
And I can eventually place it as a submodule of pyside2-setup, which does not hurt (to have it all "in a box").

**Comment by IronManMark20** [comment]
:+1: 

@ctismer, I am going to look into checking the Qt version based on the qmake given.

**Comment by thopiekar** [comment]
:+1:

**Comment by empyrical** [comment]
Can pyside.org be set up to point to pyside.github.io instead of the qt.io wiki redirect that happens right now?

**Comment by ctismer** [comment]
@empyrical no idea, I'm only in the github PySide teams.

**Comment by ctismer** [comment]
@IronManMark20 et.al. maybe you are interested, I started a little qmake parser, because I thought I need to analyze a certain setting. I had no good place for it, and it is not clear what it should do.

You find it in the wiki, qmake_miniparser.py (only visible when checked out)

**Comment by empyrical** [comment]
you can also get it online like this too:

https://raw.githubusercontent.com/wiki/PySide/pyside2/qmake_miniparser.py

**Comment by IronManMark20** [comment]
Why not just check the call of `qmake --version`? Then parse the output of that. When I run it I get:
```
Qt: Untested Windows version 10.0 detected!
QMake version 3.0
Using Qt version 5.4.1 in C:/Qt/Qt5.4.1/5.4/mingw491_32/lib
```

**Comment by ctismer** [comment]
Well, this was just a starter, with proper include file handling.
It needs a bit more, like $$variable expansion and maybe a good interface
as a module that you can ask some simple questions, instead of running the parser...

I just thought to keep it, maybe even make a PyPI module, just to have simple access to Qt settings.
But this is not so much needed, right now. If you motivate me by a proposal, I will of course
continue.

Cheers - Chris

**Comment by empyrical** [comment]
So it now compiles nicely on Ubuntu 15.04, and also successfully imports aswell!

![](https://i.imgur.com/BqW6R2X.png)

Using Qt 5.4.1 and Python 3.4.3, and this (adapted from the wikipedia example of PySide)

```python
import sys

import PySide2
from PySide2 import QtWidgets

app = QtWidgets.QApplication(sys.argv)

win = QtWidgets.QLabel()

win.resize(320, 240)  
win.setText("Hello from PySide " + PySide2.__version__ + "!") 
win.setWindowTitle("Hello, world!") 
win.show()

sys.exit(app.exec_())
```
There's a bit of an oddity here though: 2/5 times I try launching the script, it will segfault at the point where it's trying to resize the window.

**Comment by ctismer** [comment]
Howdy,

it is a wonder that you got this far.
Please note:

- any use of QtWidgets is flawed.
We need to adjust the version of the ABI.

Please try again, using a debug build. Then you will see why it crashes
when.

cheers - Chris


On 24.09.15 02:45, empyrical wrote:
>
> So it now compiles nicely on Ubuntu 15.04, and also successfully
> imports aswell!
>
> <https://camo.githubusercontent.com/30279b408f32289fbfe8dc3f84b23a3c1dd91435/68747470733a2f2f692e696d6775722e636f6d2f427157365232582e706e67>
>
> Using Qt 5.4.1 and Python 3.4.3, and this (adapted from the wikipedia
> example of PySide)
>
> import sys
>
> import PySide2
> from PySide2 import QtWidgets
>
> app = QtWidgets.QApplication(sys.argv)
>
> win = QtWidgets.QLabel()
>
> win.resize(320, 240)  
> win.setText("Hello from PySide " + PySide2.__version__ + "!") 
> win.setWindowTitle("Hello, world!") 
> win.show()
>
> sys.exit(app.exec_())
>
> There's a bit of an oddity here though: 2/5 times I try launching the
> script, it will segfault at the point where it's trying to resize the
> window.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PySide/pyside2/issues/2#issuecomment-142770034>.
>


-- 
Christian Tismer             :^)   tismer@stackless.com
Software Consulting          :     http://www.stackless.com/
Karl-Liebknecht-Str. 121     :     https://github.com/PySide
14482 Potsdam                :     GPG key -> 0xFB7BEE0E
phone +49 173 24 18 776  fax +49 (30) 700143-0023

**Comment by empyrical** [comment]
Built it again using `--debug` on the debug build of python, and here's the GDB stacktrace of a crash:

https://gist.github.com/empyrical/36e03b4bb08b85bf4098

I don't know a whole lot about GDB yet so if there's anything else you think i should run on the coredump I can do that too.

Judging by this stacktrace, would it be reasonable to assume that PySide is running into problems with the garbage collector?

**Comment by ctismer** [comment]
Hi,

the coredump does not help, since we know the data structure is wrong.

Please run this and you should see an assertion:

from PySide2 import QtWidgets
app=QtWidgets.QApplication(())
w=QtWidgets.QLabel("hello")
w.show() # here it asserts and crashes

**Comment by thopiekar** [comment]
For those who are interested and got Ubuntu Vivid or Wily installed:
I'm working currently on a PPA with nightly builds of PySide2.

*Shiboken2 is already built - https://code.launchpad.net/~thopiekar/+recipe/shiboken2-daily
*PySide2 is on the go - https://code.launchpad.net/~thopiekar/+recipe/pyside2-daily

* See: https://code.launchpad.net/~thopiekar/+archive/ubuntu/pyside-git

Happy hacking//testing

**Comment by empyrical** [comment]
I've built PySide2 in debug mode against a build of Qt that was build with `-debug`, but am not seeing any assertions - just `Segmentation fault (core dumped)` when it crashes, is there anything else I need to do to be able to see assertions?

Also, I've noticed that doing nothing but importing `QtCore` then quitting the interpreter is enough to get a segault (it happens just as the interpreter is exiting)

I'll have to see if this happens for me in your builds, @thopiekar

**Comment by ctismer** [comment]
are you sure you have a debug python build? PySide2-setup claims debug, even if it is a release python. There is just a warning. Well, this could all be a Linux thing, I tested on OS X and 5.5

**Comment by thopiekar** [comment]
@empyrical, sadly I've got trouble with this tool: http://bazaar.launchpad.net/~thopiekar/+junk/pyside2-packaging/view/head:/dh_pycmakedeps
It came with the pyside(1) packaging on Ubuntu/debian and it throws an useless error message + it's written in Perl, so I've got no clue what this (ugly) code does :angry:

**Comment by thopiekar** [comment]
@empyrical Made many fixes to the packaging - they will be available soon.
The current packages might miss some modules. More fixes with right package
version and description will follow tomorrow.
Am 26.09.2015 17:58 schrieb "Christian Tismer" <notifications@github.com>:

**Comment by ctismer** [comment]
Hi friends! What happened that made everybody so quiet? Did I close too many issues/threads?

**Comment by IronManMark20** [comment]
@ctismer no, not at all. Mostly I have been very busy recently, I plan on getting back to doing CI work this week.

**Comment by jumarock** [comment]
@ctismer, I have been busy too. Had you a chance to work on the patches related to the new qt5 metaobject ? In the coming days I will try to find time to build in debug mode (VS 2010) + test the signal/slots.

**Comment by ctismer** [comment]
On 19.10.15 18:05, jumarock wrote:
>
> @ctismer <https://github.com/ctismer>, I have been busy too. Had you a 
> chance to work on the patches related to the new qt5 metaobject ? In 
> the coming days I will try to find time to build in debug mode (VS 
> 2010) + test the signal/slots.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/PySide/pyside2/issues/2#issuecomment-149259957>.
>

Hi,

I was very busy, but am trying to apply the patches.
A few are remaining, and I have one day left until I'm gone for October.

cheers - Chris

-- 
Christian Tismer             :^)   tismer@stackless.com
Software Consulting          :     http://www.stackless.com/
Karl-Liebknecht-Str. 121     :     https://github.com/PySide
14482 Potsdam                :     GPG key -> 0xFB7BEE0E
phone +49 173 24 18 776  fax +49 (30) 700143-0023

**Comment by empyrical** [comment]
I looked at a PySide crash with the XCode debugger, and `QWidgetWindow::updateObjectName()` seems to be where the crashes are happening when calling `show()` on a widget; this is the body of that function:

```c++
void QWidgetWindow::updateObjectName()
{
    QString name = m_widget->objectName();
    if (name.isEmpty())
        name = QString::fromUtf8(m_widget->metaObject()->className()) + QStringLiteral("Class");
    name += QStringLiteral("Window");
    setObjectName(name);
}
```

I think the main problem here is that PySide's QObject wrapper doesn't have a working `metaObject()` method yet. If you use `setObjectName` on a widget before calling `show` on it, it will get past this function fine. But if you don't, it will attempt to make a name based off of `m_widget->metaObject()->className()` and the crash happens (probably) because nothing's there.

If you give it a name, it will move on to crashing when something tries to call `QApplication::font` on the widget, which also tries to get the className from `metaObject()` *Hopefully*, if `metaObject()` is fixed, this will fix a bunch of crashes.

**Comment by jumarock** [comment]
I'm working on a local branch that handles the Qt5 metaobject modifications. So far, I'm able to display widget without crash and the signals/slots seems to be functional.
I will see with christian how to proceed to validate theses changes.

**Comment by IronManMark20** [comment]
@jumarock , can't wait to try that out.

also, for everyone else, I have been having some trouble getting the Qt dependancy part to install properly on Appveyor. I will try to work on it some more.

@The-Compiler, if you have the time, could you start doing a Travis CI? Seems you have some experience in this...

**Comment by The-Compiler** [comment]
I can try, though it might take me a while as I'm quite busy at the moment.

I got lost a bit - are there some build instructions or your work-in-progress `appveyor.yml` anywhere so I have a starting point?

**Comment by IronManMark20** [comment]
@The-Compiler Since appveyor is pretty different, I am not sure if it would cross over. As I said, it isn't done, either. Basically, grab the [dependencies](https://github.com/PySide/pyside2/wiki/Dependencies#ubuntu), then build, pointing to the dependencies. (The readme has the needed flags to point to deps.)

**Comment by ctismer** [comment]
@jumarock I will finish the merge on Saturday 7th, hopefully. Right now I'm working in a branch. Your patches are in the wiki repo of pyside2-setup, where I filtered redundant things out etc. That repo is only visible after a local checkout.

**Comment by ctismer** [comment]
I was thinking to close this long thread, again, but somehow the discussion should stay, somewhere.
Any ideas?

**Comment by empyrical** [comment]
[Gitter](https://gitter.im/) might be a good place to move general discussion

**Comment by IronManMark20** [comment]
@empyrical , we already have a gitter for internal chat (@ctismer, could you invite @empyrical ?), but lets just say it is not the most frequented (yet). I normally am available for chat on gitter.

**Comment by empyrical** [comment]
Could make a public gitter and set up a gitter badge in the PySide readme so more people see it

**Comment by ctismer** [comment]
I want to take care about that, because people really ask for such a facility.
Please remind me, I only have time this weekend.

**Comment by ctismer** [comment]
Failed to do that, again. Maybe in the evening. Do we really want a public gitter?

**Comment by empyrical** [comment]
Sorry for not reminding, didn't check github for a few days! I personally still think it'd be useful

**Comment by ctismer** [comment]
Hi @empyrical I tried to add you to the gitter group https://gitter.im/PySide/PySide2-devel#

**Comment by empyrical** [comment]
I seem to get a 404 when going to that link, maybe try inviting me again? I haven't used private gitter rooms before, so not sure what needs to be done to get them working

**Comment by ctismer** [comment]
Ok, I have removed the invite and issued it again. Good luck ;-)

**Comment by empyrical** [comment]
Hmm, when you made the room did you set it to "private" or for "all org members"? If the latter, you might need to make a new room set to "private" instead so I can see it.

**Comment by ctismer** [comment]
I did not make the room. It was made by one of you, long ago.

**Comment by empyrical** [comment]
Odd. In that case, I do not know how to retrieve/modify the current permissions of a room.

In any case, I've typed up a quick little walkthrough to getting a general PySide2 room up:

 * Hop over to Gitter.im
 * Log in, and in the bottom left corner click the "Create A Room" button
 * Select the "Repository" option
 * Select `PySide/pyside2` from the repository list
 * If you get a permissions pop up, make sure to select "Public"
 * It should also ask you if you want to add a badge via a pull request, if you don't want to do that paste this in the README to get the chat badge:  
```
[![](https://img.shields.io/gitter/room/PySide/pyside2.svg)](https://gitter.im/PySide/pyside2)
```

Should hopefully alleviate the need for this cartoonishly big issue.

**Comment by ctismer** [comment]
Ok there is a new public gitter room: https://gitter.im/PySide/pyside2
I guess this thread should stop here.

p.s.: is there a way to export this thread into Gitter? Or should I keep it open?

**Comment by The-Compiler** [comment]
I won't participate there, unless the IRC bridge (which they call "still very much beta") actually works. I think the most things on [Please don't use Slack for FOSS projects](https://drewdevault.com/2015/11/01/Please-stop-using-slack.html) also hold true for gitter.

Also, I *tried* actually writing something in that chat, and nothing happened... fun.

**Comment by empyrical** [comment]
If you close it, people will still be able to comment in it if need be - maybe just a link to this thread could be posted in the Gitter so people looking in the history will find it.

I definitely see your reasoning, @The-Compiler. Almost every single historical repo, website, and bug tracker link I've stumbled across is completely dead now and moving to something more open later might be better in the future once PySide is off the ground.

I personally like Gitter because everyone doesn't need to be online to see messages that have been posted unlike IRC (unless you set up a bouncer), and a message board would scratch that itch too. How do you feel about a Discourse board or something? I'm personally a huge fan of that message board system.

On an unrelated note, how hard to you think it might be to add PySide2 support to your QuteBrowser? I think running full apps like that might be a good way to test PySide2.

**Comment by The-Compiler** [comment]
Ironically, I'd prefer a mailinglist over a discourse board :wink: I'm involved in too much stuff for... let me call it pull-based communication to work. Much better when stuff lands in my inbox (or IRC client), then I'll actually read it.

I'm eager to try out PySide2 in qutebrowser (also see https://github.com/The-Compiler/qutebrowser/issues/1076), but right now I'm in the middle of some big things I need to tackle, so it's unlikely I'll get to this in the next months (but I'd of course welcome contributions!).

The test suite is getting better and better (with 74% coverage currently), so that'll sure help to see if everything is alright.
